### DISCUSSION TITLE
Add support for a type-safe routing

### DISCUSSION THREAD

**Discussion by jferrettiboke** [expo/router#51] [Ideas]
It'd be great to add support for a type-safe routing. Something similar to [nextjs-routes](https://github.com/tatethurston/nextjs-routes).

**Comment by EvanBacon** [comment]
We've talked about doing this. Will probably not get around to it until the beta has been completed.

**Reply by Zeko369** [reply]
Would you be interested in contributions for this?

**Reply by EvanBacon** [reply]
Seems like a difficult and opinionated project that will probably be implemented using file watching and AST parsing in Expo CLI. Probably best if I handle it personally, but feel free to give it a shot.

**Reply by sandulat** [reply]
@EvanBacon Related https://github.com/expo/router/discussions/448
